src/Native/configure.cmake

```diff
@@ -196,13 +199,53 @@ check_function_exists(
     kqueue
     HAVE_KQUEUE)
 
+check_cxx_source_compiles(
+     "
+     #include <sys/types.h>
+     #include <netdb.h>
+
+     int main()
+     {
+         const void* addr;
+         socklen_t len;
+         int type;
+         struct hostent* result;
+         char* buffer;
+         size_t buflen;
+         struct hostent** entry;
+         int* error;
+         gethostbyaddr_r(addr,  len, type, result, buffer, buflen, entry, error);
+         return 0;
+     }
+     "
+     HAVE_GETHOSTBYADDR_R)
+
+check_cxx_source_compiles(
+     "
+     #include <sys/types.h>
+     #include <netdb.h>
+
+     int main()
+     {
+         const char* hostname;
+         struct hostent* result;
+         char* buffer;
+         size_t buflen;
+         struct hostent** entry;
+         int* error;
+         gethostbyname_r(hostname, result, buffer, buflen, entry, error);
+         return 0;
+     }
+     "
+     HAVE_GETHOSTBYNAME_R)
+
 check_function_exists(
-    gethostbyname_r
-    HAVE_GETHOSTBYNAME_R)
+    getaddrinfo
+    HAVE_GETADDRINFO)
 
 check_function_exists(
-    gethostbyaddr_r
-    HAVE_GETHOSTBYADDR_R)
+    getnameinfo
+    HAVE_GETNAMEINFO)
```

Comment:
> Should we change the preprocessor conditions in code to #ifdef __NETBSD__

No, but it seems like there are currently three cases:
- Do we have a thread safe gethostbyname/gethostbyaddr (HAVE_THREAD_SAFE_GETHOSTBYNAME_AND_GETHOSTBYADDR)?
- If not, do we have gethostbyname_r?
- If not, we need a temporary stub to fill in later that at least allows compilation to be successful and that returns an error at run time.

Is that correct?  If so, why do we need any more feature detection than we already have?